### PR TITLE
fix(ZNTA-2015): Allow admin to change a user's name

### PR DESCRIPTION
**Pull request by djansen-redhat**
JIRA issue URL: https://zanata.atlassian.net/browse/ZNTA-2015

Add Full Name field, editable.
Checks for >0 visible characters
Provides a placeholder of the original name.
Limits to 80 characters.
Also checks the register page for whitespace.

## Reviewer tasks

The following is based on the
[Zanata Development Guidelines](https://github.com/zanata/zanata-platform/wiki/Development-Guidelines).
As a reviewer, you should check the guidelines regularly for updates, and just
to refresh your memory.

The reviewer can use this list for reference to make sure
they have considered all the important points.

Use the checkboxes or not, whatever works best for you.

 - Code quality
   - [ ] Code passes style check (checkstyle/eslint)
   - [ ] Code is clear and concise
   - [ ] UI code is internationalised
   - [ ] Code has adequate comments where appropriate
   - Code is in an appropriate place
     - [ ] Classes are in appropriate packages
     - [ ] Code fits with class's responsibilities (SRP)
   - [ ] Configuration files are documented enough
   - If code is removed
     - [ ] No remaining code references the removed code
       - [ ] No orphaned rewrite rules
       - [ ] No orphaned config settings
     - [ ] No remaining comments refer to the removed code
     - [ ] Unused imports and libraries are removed
 - Testing
   - [ ] New code has tests
   - [ ] Changed code has tests
   - [ ] All tests pass
   - [ ] Test coverage stable or increased
 - Documentation
   - [ ] New features are documented
   - [ ] Docs removed for deleted features
   - [ ] Docs updated for all changed features
   - [ ] Developer/sysadmin docs updated if appropriate
 - If there are new dependencies
   - [ ] Does each new dependency pass all the
         [Considerations for new dependencies](https://github.com/zanata/zanata-platform/wiki/Development-Guidelines#new-technologies-and-dependencies)?

----
*This template can be updated in .github/PULL_REQUEST_TEMPLATE.md*

---
This change is [Reviewable](https://reviewable.io/reviews/zanata/zanata-platform/349)

### CODE REVIEW

**Comment by aeng** [comment]
Reviewed 5 of 6 files at r1.
Review status: 5 of 6 files reviewed at latest revision, 1 unresolved discussion, some commit checks failed.

---

*[server/zanata-war/src/main/java/org/zanata/action/UserAction.java, line 167 at r1](https://reviewable.io:443/reviews/zanata/zanata-platform/349#-KlRpU8djdfp44ut_WDt:-KlRpU8djdfp44ut_WDu:b9t5tec) ([raw file](https://github.com/zanata/zanata-platform/blob/66f34f3c10b3ce861aa83afd4c46702cfe968b08/server/zanata-war/src/main/java/org/zanata/action/UserAction.java#L167)):*
> ```Java
>         }
>         String newName = name.trim();
>         if (!newName.isEmpty() && !newName.equals(originalName)) {
> ```

Can use StringUtils.equals, that include empty and null checker.

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/zanata/zanata-platform/349)*

**Comment by aeng** [comment]
:lgtm:

---

Reviewed 2 of 2 files at r2.
Review status: all files reviewed at latest revision, 1 unresolved discussion.

---



*Comments from [Reviewable](https://reviewable.io:443/reviews/zanata/zanata-platform/349#-:-KlRyEhjo4DecTvx74BS:bnfp4nl)*

File: server/functional-test/src/main/java/org/zanata/page/administration/ManageUserAccountPage.java [hunk 1]
```diff
@@ -52,6 +53,17 @@ public ManageUserAccountPage(WebDriver driver) {
...
+        return new ManageUserAccountPage(getDriver());
+    }
+
+    public String getCurrentName() {
+        log.info("Query user's name");
+        return readyElement(fullNameField).getAttribute("value");
+    }
+
     public ManageUserAccountPage enterPassword(String password) {
         log.info("Enter password {}", password);
```
Comment (seanf) [inline server/functional-test/src/main/java/org/zanata/page/administration/ManageUserAccountPage.java]: Seems like the above log messages should be debug, not info. And this (old) log message shouldn't include a password.

**Comment by seanf** [comment]
Review status: all files reviewed at latest revision, all discussions resolved, some commit checks broke.

---

*[server/functional-test/src/main/java/org/zanata/page/administration/ManageUserAccountPage.java, line 68 at r2](https://reviewable.io:443/reviews/zanata/zanata-platform/349#-KlRfxBm4y9NvzXroBNP-r2-68:-KlXA0slJHvM_WOo_9yt:b-dsmwi6) ([raw file](https://github.com/zanata/zanata-platform/blob/b363d3b5854ebf79bc6a87586ed85c14b19f598e/server/functional-test/src/main/java/org/zanata/page/administration/ManageUserAccountPage.java#L68)):*
<details><summary><i>Previously, seanf (Sean Flanigan) wrote…</i></summary><blockquote>

Seems like the above log messages should be debug, not info. And this (old) log message shouldn't include a password.
</blockquote></details>

Disregard: it's only logging test data, since this is test code. log.info makes sense too, for a test.

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/zanata/zanata-platform/349)*

**Comment by djansen-redhat** [comment]
Codecov is +0.04%.

**Pull request merged**